### PR TITLE
fix(table-bloat): monotonic stale-vacuum thresholds and severity escalation

### PR DESCRIPTION
**Pull request by emancu**
## Summary

Two related fixes to the `table-bloat` check, both around severity correctness:

**1. `stale-vacuum` thresholds were non-monotonic.** FAIL fired at dead>50K after 7 days while WARN required dead>100K after 3 days — a ~60K-dead table could FAIL without ever qualifying for WARN, and neither tier considered dead-tuple percentage. New predicates (FAIL is a strict subset of WARN on every axis):

- **WARN**: `((dead% ≥ 10 AND dead ≥ 10K) OR dead ≥ 100K) AND last vacuum > 3d`
- **FAIL**: `((dead% ≥ 10 AND dead ≥ 400K) OR dead ≥ 1M) AND last vacuum > 12d`
- **never-vacuumed**: treated as infinite vacuum age; FAIL at dead ≥ 250K

The FAIL percentage arm stays at 10% (not higher) because autovacuum's default scale factor triggers around 20% dead — sustained ≥10% with a large absolute count and 12+ days without vacuum means autovacuum is falling behind, which is the condition worth paging on. The 10K floor on the percentage arms keeps tiny tables from warning on percentage alone.

Measured against a 120-database production corpus: the old FAIL fired on ~89 tables across 35 databases; the new FAIL fires on ~8 tables across 6 databases.

**2. Finding severity now escalates with critical rows.** All three sub-findings (`high-dead-tuples`, `stale-vacuum`, `large-bloated-tables`) hard-coded finding severity WARN while pushing FAIL-severity rows — so red rows rendered under a yellow WARN header and the report never escalated to FAIL. Finding severity is now FAIL when any critical row exists, restoring the row-severity ≤ finding-severity invariant.

## Testing

- 10 boundary cases over the new predicate matrix + severity-escalation tests for all three sub-findings
- `go build ./...`, `go test -count=1 ./...` all green, `golangci-lint run` 0 issues, codegen freshness verified

### CODE REVIEW

File: checks/tablebloat/check.go [hunk 1]
```diff
@@ -71,6 +71,15 @@ func (c *checker) Check(ctx context.Context) (*check.Report, error) {
...
 }
 
+// findingSeverity keeps the finding at least as severe as its worst row, so
+// the report escalates to FAIL when critical rows exist.
+func findingSeverity(critical []db.TableBloatRow) check.Severity {
+	if len(critical) > 0 {
+		return check.SeverityFail
+	}
+	return check.SeverityWarn
+}
```
Comment (emancu) [inline checks/tablebloat/check.go]: 🤔 Isnt this the same as iterate over ALL rows and keepoing max?

Also, why does the check do this, and not the printer? I dont get it.

File: checks/tablebloat/check.go [hunk 2]
```diff
@@ -135,7 +144,7 @@ func checkHighDeadTuples(rows []db.TableBloatRow, report *check.Report) {
 	report.AddFinding(check.Finding{
 		ID:       "high-dead-tuples",
 		Name:     "Dead Tuple Percentage",
-		Severity: check.SeverityWarn,
+		Severity: findingSeverity(critical),
```
Comment (emancu) [inline checks/tablebloat/check.go]: 🤔 I dont think this is correct. This should remain WARN.
Even if its concerning, this wont prevent the DB to run or anything. Performance will be impacted, but not sure if i want to go that bad.

Unless this is on purpose, why is this severity so high?

File: checks/tablebloat/check.go (hunk 1 above)
Comment (emancu) [reply checks/tablebloat/check.go]: Yes — it was max-over-rows in disguise. Replaced with an explicit `maxRowSeverity(rows)` (floored at WARN) in f8010a9 so the derivation is visible. On check vs printer: severity is classification, and `Report.Severity` = max over findings feeds JSON/scorecards too — if a printer computed it, only the CLI would see the escalation and every other consumer would under-report. Deriving from rows also makes the row ≤ finding invariant hold by construction.

File: checks/tablebloat/check.go (hunk 2 above)
Comment (emancu) [reply checks/tablebloat/check.go]: Agreed — demoted in f8010a9: high-dead-tuples is now hard WARN, and the former >40% FAIL rows collapse into the single >20% WARN bucket (a FAIL row under a WARN finding would break the row ≤ finding invariant). stale-vacuum keeps its FAIL tier per the approved 12d/1M predicates. Note: large-bloated-tables still escalates on >10GB && >20% dead — same chronic shape as this one, and it has no disk-headroom context to justify paging. Want it demoted to WARN-only too (separate follow-up)?
